src/gamestate.ts

```diff
@@ -1,7 +1,59 @@
+import { areCoordinatesValid } from "./user";
+import { Coordinates } from "./utils";
+
 export interface GameState {
   board: Array<Array<Cell>>;
 }
 
 export interface Cell {
   value: "empty" | "black";
 }
+
+export function generateNewBoard(): GameState {
+  return {
+    board: [
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+    ],
+  };
+}
+
+export function updateGameState(
+  previousState: GameState,
+  actionCoordinates: Coordinates
+): GameState {
+  if (!areCoordinatesValid(previousState, actionCoordinates)) {
```

Comment:
Since `areCoordinatesValid()` uses the game state and not only the user input, I feel like it would rather need to be declared in this file (`gamestate.ts`) rather than in `user.ts`. What do you think?